src/transmuxer-worker.js

```diff
@@ -207,6 +208,41 @@ class MessageHandlers {
     }, [segment.buffer]);
   }
 
+  /**
+   * Initializes the WebVttParser and passes the init segment.
+   *
+   * @param {Uint8Array} data mp4 boxed WebVTT init segment data
+   */
+  initMp4WebVttParser(data) {
+    if (!this.webVttParser) {
+      this.webVttParser = new WebVttParser();
+    }
+    const segment = new Uint8Array(data.data, data.byteOffset, data.byteLength);
+
+    // set the timescale for the parser.
+    this.webVttParser.init(segment);
```

Comment:
I'm note exactly sure how the `init` function works on the parser, but is it possible/valid to call `init` when the parser already exists and `init` has already been called?

Reply:
So, the init function for the `WebVttParser` only passes the segment data to extract a timescale. So it actually is valid to call this multiple times, especially if the timescale changes over a discontinuity. I'll add a bit to the comment here that init is ok to call repeatedly. https://github.com/videojs/mux.js/blob/e1b4d161f51782e4712d020b66d5087bd6febb3d/lib/mp4/webvtt-parser.js#L18